FWCore/Framework/test/run_PrintDependencies.sh

```diff
@@ -3,10 +3,11 @@
 # Pass in name and status
 function die { echo $1: status $2 ;  exit $2; }
 
+LOCAL_TEST_DIR="${CMSSW_BASE}/src/FWCore/Framework/test"
```

Comment:
Given that scram knows it is executing a test, couldn't we pass a special environment variable to the tests to let them know about where they come from? As seen here, that is a very useful thing for the test. It also avoids having to know exactly which CMSSW variable are available both in a local work area and in the IBs. It would also help if a test is moved from one directory to a different directory for maintenance reasons as one would not have to remember to update the internals of the test as well.

Reply:
not really, previously the [TestHelper](https://github.com/cms-sw/cmssw/blob/master/FWCore/Utilities/src/TestHelper.cc) was setting various env e.g `LOCAL_TMP_DIR, LOCAL_TEST_DIR` so when you run the unit tests executable e.g. `TestFWCoreFrameworkDeleteEarly` then it works. Of course if you run the shell script directly then yes it would have failed.

But I have no issues if tests can make use of these `SCRAM_TEST_*`

Reply:
`TestHelper` generates `LOCAL_TEST_DIR` based on the 3rd argument it is passed and that argument is embedded in the BuildFile.xml file. So executing just `TestFWCoreFrameworkDeleteEarly` should fail with an 'invalid number of arguments' error.

Reply:
ah yes, you are right